platform/genode/cai-log-client.adb

```diff
@@ -44,24 +44,28 @@ package body Cai.Log.Client is
    Red : constant String := Character'Val (8#33#) & "[31m";
    Reset : constant String := Character'Val (8#33#) & "[0m";
    Terminator : constant Character := Character'Val (0);
+   Nl : constant Character := Character'Val (10);
 
-   procedure Info (C : in out Client_Session; Msg : String)
+   procedure Info (C : in out Client_Session; Msg : String; Newline : Boolean := True)
    is
-      C_Msg : String := Msg & Terminator;
+      C_Msg : String := Msg
+                        & (if Newline then Nl & Terminator else Terminator & Terminator);
```

Comment:
I wonder, why do you append `Terminator` twice if `Newline` is false?

Reply:
Since `Nl & Terminator` is a `String` the return type of the if expression is also a String so the else branch needs to be one, too. An alternative would have been `String (1 .. 1)' (Terminator)` But appending a second null terminator works, too (It won't have any effect on the C++ code since it stops reading the string at the null terminator anyway so there could be anything).

Reply:
The most elegant solution is `(1 => Terminator)`.